cores/esp8266/core_esp8266_wiring_pwm.cpp

```diff
@@ -44,7 +44,14 @@ extern void __analogWriteFreq(uint32_t freq) {
 }
 
 extern void __analogWrite(uint8_t pin, int val) {
-  analogWriteMode(pin, val, false);
+  if (pin > 16) {
```

Comment:
This changed standard behavior.  `analogWrite()` in  Arduino-land doesn't allow open drain in the cores I've looked at, so this should be undone.  The user has to explicitly use the overloaded call we added in this repo a while back (it's for a specific closed bug).

Reply:
Case in point - GitHub review is completely unreadable here, I'm have the hardest time figuring out which line you are referring to with your remark :-)

Reply:
The intent is a bit different - `analogWrite` before PR always resets the open drain mode, this change simply prevents that. Otherwise, the overloaded call sows more confusion than just fixing an issue.

Reply:
I explained in https://github.com/esp8266/Arduino/pull/7841